### PR TITLE
Fix: When no results in search editor it throws an error

### PR DESCRIPTION
**Pull request by janajar**
Fixes issue #233200 
Added a null check for matchRange in the iterateThroughMatches function to ensure setSelection is only called with valid arguments.

Authored with @aayyob

### CODE REVIEW

**Comment by janajar** [comment]
@microsoft-github-policy-service agree

**Review by 818Nawaf — APPROVED** [review]

**Comment by janajar** [comment]
Hi @roblourens,
This is ready to merge once you approve it.

**Review by roblourens — APPROVED** [review]
Thanks!

**Review by sandy081 — APPROVED** [review]

**Pull request merged**
